app/controllers/spree/orders_controller.rb

```diff
@@ -25,8 +25,8 @@ class OrdersController < ::BaseController
     before_action :check_at_least_one_line_item, only: :update
 
     def show
-      ProcessPaymentIntent.new(params["payment_intent"], params[:id]).call!
       @order = Spree::Order.find_by!(number: params[:id])
+      ProcessPaymentIntent.new(params["payment_intent"], @order).call!
```

Comment:
It turns out that I did that for a reason, but couldn't remember what it was when I saw that you'd switched it :-D

c851705d5a6160f9c77249d5a747d41587fc8151

In order for the order to show up as "Paid" on the page that we redirect to, we have to load it from the database **after** processing the payment intent.

Reply:
:+1: I prefer to see a `reload`. To me, it conveys what you explained.